src/services/tracker.js

```diff
@@ -3,21 +3,23 @@ export default class Tracker {
   /**
    * Tracker mixin for aggregating service data.
    * @param {riot} riot - Riot reference.
-   * @param {Array} services - Collection of services for data aggregration.
    */
   constructor(riot, services) {
     this.$riot = riot
     this.$riot.observable(this)
-    this.update(services)
+    this.update()
   }
 
   /**
    * Update tracker service data.
-   * @param {Array} services - Services
    */
-  update(services) {
-    this.services = services
-    this.trigger('update')
+  update() {
+    this.services = []
+
+    fetch('/api/services').then(response => response.json()).then(json => {
+        this.services = json
+        this.trigger('update')
```

Comment:
Not a big deal but we can probably dispatch another observable event if the request fails, even if it's not going to be exercised in the app this would probably be a nice example for the book. What are your thoughts?